src/Uno.UI/UI/Xaml/FrameworkElement.EffectiveViewport.ViewportInfo.cs

```diff
@@ -0,0 +1,103 @@
+﻿#nullable enable
+
+using System;
+using System.Linq;
+using Windows.Foundation;
+using Uno.Extensions;
+
+namespace Windows.UI.Xaml
+{
+	// Internal interface used to allow communication between the real FrameworkElement
+	// and presenters that are only implementing the IFrameworkElement interface (cf. FrameworkElementMixins.tt).
+	// It must not be used anywhere else out of the file.
+	internal interface IFrameworkElement_EffectiveViewport
+	{
+		void InitializeEffectiveViewport();
+		IDisposable RequestViewportUpdates(IFrameworkElement_EffectiveViewport? child = null);
+		void OnParentViewportChanged(IFrameworkElement_EffectiveViewport parent, ViewportInfo viewport, bool isInitial = false);
+		void OnLayoutUpdated();
+	}
+
+	internal struct ViewportInfo : IEquatable<ViewportInfo>
```

Comment:
Why a struct ?

Reply:
Using a `class` for this would most probably be overkill, we could have used a `record`, but this is designed to be a "super `Rect`" that is used as a drop-in replacement in some cases.